tests/unit_tests/data_model/test_validate_data.py

```diff
@@ -771,3 +771,56 @@ def test_validate_data_dict_using_json_schema(caplog):
     with caplog.at_level(logging.DEBUG):
         data_validator._validate_data_dict_using_json_schema(valid_data, None)
     assert "Skipping validation of dict type" in caplog.text
+
+
+def test_get_value_and_units_as_lists():
+    data_validator = validate_data.DataValidator()
+
+    # Test with single value and unit
+    data_validator.data_dict = {"value": 100, "unit": "m"}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100]
+    assert units == ["m"]
+
+    # Test with list of values and units
+    data_validator.data_dict = {"value": [100, 200], "unit": ["m", "cm"]}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100, 200]
+    assert units == ["m", "cm"]
+
+    # Test with numpy array of values and units
+    data_validator.data_dict = {"value": np.array([100, 200]), "unit": np.array(["m", "cm"])}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100, 200]
+    assert units == ["m", "cm"]
+
+    # Test with unit as "null"
+    data_validator.data_dict = {"value": [100, 200], "unit": ["m", "null"]}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100, 200]
+    assert units == ["m", None]
+
+    # Test with mixed types
+    data_validator.data_dict = {"value": [100, 200], "unit": np.array(["m", "cm"])}
+    values, units = data_validator._get_value_and_units_as_lists()
+    assert values == [100, 200]
+    assert units == ["m", "cm"]
+
+
+def test_validate_value_and_unit_for_dict(reference_columns):
+    data_validator = validate_data.DataValidator()
+    data_validator._data_description = reference_columns
+
+    # Test case for dict type
+    data_validator.data_dict = {"value": {"key": "value"}, "unit": None, "type": "dict"}
```

Comment:
Shouldn't the unit here be `"null"`? I don't remember which of the options we decided to stick with, but either way it is inconsistent with the example in the tests above.

Reply:
You are correct, we are using everywhere "null", so I've added a test for "null".

I want to keep the test for `None`, as the "null" is used solely because of the usage of string-type arrays (as in sim-telarray). I hope that on a medium time scale, we can move away from these strings.